src/service.rs

```diff
@@ -10,9 +10,7 @@ use futures::{Future, Poll};
 /// This is not intended to be implemented directly. Instead, it is a trait
 /// alias of sorts. Implements the `tower_service::Service` trait using
 /// `http::Request` and `http::Response` types.
-pub trait HttpService: ::sealed::Sealed {
-    /// Request payload.
-    type RequestBody: Body;
+pub trait HttpService<RequestBody>: ::sealed::GenericSealed1<RequestBody> {
```

Comment:
Hmm, we don't *have* to also change this trait to match the new `Service<Request>` style; we could just make the change internally. Is there a particularly good reason for why `RequestBody` should be turned into a generic type argument on `HttpService`?

Reply:
Also, this no longer requires that `RequestBody: Body`. Is that intentional?

Reply:
@jonhoo 
> Hmm, we don't *have* to also change this trait to match the new `Service<Request>` style 

I believe we _do_ need to change this trait to be generic. Otherwise, in the blanket impl of `HttpService` for `T: Service<...>`, the body type parameter is unbounded.

This code:
```rust
impl<T, B1, B2> HttpService for T
where T: Service<Request<B1>,
                Response = Response<B2>>,
      B1: Body,
      B2: Body,
{
    type RequestBody = B1;
    type ResponseBody = B2;
    // ...
}
```
fails to compile with this error:
```
error[E0207]: the type parameter `B1` is not constrained by the impl trait,self type, or predicates
  --> src/service.rs:66:9
   |
66 | impl<T, B1, B2> HttpService for T
   |         ^^ unconstrained type parameter
```

> Also, this no longer requires that `RequestBody: Body`. Is that intentional?

No, that's an oversight on my part. Thanks for catching it!

Reply:
Put the `RequestBody: Body` bound back in d27d22a619c8f204caa6c3bef935d99267281597.

Reply:
Hmm, interesting. I guess it does have to be that way, it just doesn't immediately make sense to me why that doesn't compile. Oh well, I guess we're changing `HttpService` too then :/

Reply:
Apparently (https://github.com/tower-rs/tower/pull/109#discussion_r228031054) the type system may support this post-chalk.